GVFS/GVFS.UnitTests/Common/ProductUpgraderInfoTests.cs

```diff
@@ -0,0 +1,79 @@
+using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using GVFS.Tests.Should;
+using GVFS.UnitTests.Mock.Common;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.UnitTests.Common
+{
+    [TestFixture]
+    public class ProductUpgraderInfoTests
+    {
+        private Mock<PhysicalFileSystem> mockFileSystem;
+        private ProductUpgraderInfo productUpgraderInfo;
+        private string upgradeDirectory;
+        private string expectedNewVersionExistsFileName = "HighestAvailableVersion";
+        private string expectedNewVersionExistsFilePath;
+        private MockTracer tracer;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.upgradeDirectory = Paths.GetServiceDataRoot(ProductUpgraderInfo.UpgradeDirectoryName);
+            this.expectedNewVersionExistsFilePath = Path.Combine(this.upgradeDirectory, this.expectedNewVersionExistsFileName);
+            this.mockFileSystem = new Mock<PhysicalFileSystem>();
+
+            this.mockFileSystem.Setup(fileSystem => fileSystem.WriteAllText(this.expectedNewVersionExistsFilePath, It.IsAny<string>()));
+
+            this.tracer = new MockTracer();
+
+            this.productUpgraderInfo = new ProductUpgraderInfo(
+                this.tracer,
+                this.mockFileSystem.Object);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            this.mockFileSystem = null;
+            this.productUpgraderInfo = null;
+            this.tracer = null;
+        }
+
+        [TestCase]
+        public void RecordHighestVersion()
+        {
+            this.productUpgraderInfo.RecordHighestAvailableVersion(new Version("1.0.0.0"));
+            this.mockFileSystem.Verify(fileSystem => fileSystem.WriteAllText(this.expectedNewVersionExistsFilePath, It.IsAny<string>()), Times.Once());
```

Comment:
I think this is a good idea - however, as we do not read the version from this file, I will put this on my list of items to follow up on, rather than include it in this PR.

Reply:
> I will put this on my list of items to follow up on, rather than include it in this PR.

Could you create a GitHub issue that has planned follow up work items?

Reply:
You could also consider renaming the test for now to something like `CreatesHighestVersionFile` as the test is not validating that the highest version was recorded.